
# HG changeset patch
# User Ricky Stewart <rstewart@mozilla.com>
# Date 1602002309 0
# Node ID db9bdaed1b96ac233e1e04e56b542965918c253a
# Parent  0794364628877460ff5896614b3b993f3ab55400
Bug 1669471 - Get rid of `DeprecationWarning` in `file_generate.py` r=firefox-build-system-reviewers,mhentges

`imp` is deprecated since Python 3.4 and later Python versions are very noisy about printing `DeprecationWarning`s; instead, use its replacement, `importlib`.

Differential Revision: https://phabricator.services.mozilla.com/D92632

diff --git a/python/mozbuild/mozbuild/action/file_generate.py b/python/mozbuild/mozbuild/action/file_generate.py
--- a/python/mozbuild/mozbuild/action/file_generate.py
+++ b/python/mozbuild/mozbuild/action/file_generate.py
@@ -4,17 +4,17 @@
 
 # Given a Python script and arguments describing the output file, and
 # the arguments that can be used to generate the output file, call the
 # script's |main| method with appropriate arguments.
 
 from __future__ import absolute_import, print_function
 
 import argparse
-import imp
+import importlib.util
 import os
 import six
 import sys
 import traceback
 
 from mozbuild.pythonutil import iter_modules_in_path
 from mozbuild.makeutil import Makefile
 from mozbuild.util import FileAvoidWrite
@@ -52,19 +52,19 @@ def main(argv):
     # the script as:
     #
     #    python script arg1...
     #
     # then importing modules from the script's directory would come for free.
     # Since we're invoking the script in a roundabout way, we provide this
     # bit of convenience.
     sys.path.append(os.path.dirname(script))
-    with open(script, 'r') as fh:
-        module = imp.load_module('script', fh, script,
-                                 ('.py', 'r', imp.PY_SOURCE))
+    spec = importlib.util.spec_from_file_location('script', script)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
     method = args.method_name
     if not hasattr(module, method):
         print('Error: script "{0}" is missing a {1} method'.format(script, method),
               file=sys.stderr)
         return 1
 
     ret = 1
     try:

